src/library/basesqltablemodel.h

```diff
@@ -139,6 +139,23 @@ class BaseSqlTableModel : public QAbstractTableModel, public TrackModel {
         }
         int m_column;
         Qt::SortOrder m_order;
+        
+        friend QTextStream& operator >> (QTextStream &in, SortColumn &sc) {
```

Comment:
> I am afraid we hit a conceptual issue. :-/ 
> The current solution is a simple serialize without ans sanity checks. The same we have with the header state.
> 
> Since this is not that critical data, that "could" work as it is, including all issues we already heave.

I would prefer to treat user data as "sacred" -- if we decide to save something we should save it in a future-proof manner or not save it at all :). 

> If not, we should should probably ditch the SettingsDAO entirely and move to mixxx.cfg, which allows us to parse and check the values before using. .. and is human readable.

SettingsDAO and mixxx.cfg don't seem any different in this regard, are they? You can store any QString / QByteArray in either of them. It's up to the code that reads/stores to it to parse/check. 

We could use a protocol buffer to serialize column sort state -- we could store that in either mixxx.cfg or SettingsDAO and we would get parsing, serializing and the ability to migrate users to a future format easily. Human readability can be achieved with protobuf text formatting but I'm not sure if it needs to be human readable?